### PR TITLE
Only apply DATABASE_URL for Rails.env

### PR DESCRIPTION
**Pull request by matthewd**
As we like ENV vars, also support DATABASE_URL_#{env}, for more obscure use cases.

This needs more tests:
- There's nothing about the 'DATABASE_URL_FOO' handling
- And nothing showing that a normally-configured "foo" is unaffected by the DATABASE_URL (unless our current env is called 'foo')

Also, the `config` method could probably look substantially less awful.
#14616 // @rafaelfranca

### CODE REVIEW

File: railties/lib/rails/generators/rails/app/templates/config/databases/oracle.yml [hunk 1]
```diff
@@ -32,7 +32,27 @@ test:
...
+#
+#   DATABASE_URL="oracle://myuser:mypass@localhost/somedatabase"
+#
+# You can use this database configuration with:
+#
+#   production:
+#     url: <%%= ENV['DATABASE_URL'] %>
+#
 production:
-  url: <%%= ENV["DATABASE_URL"] %>
```
Comment (prathamesh-sonpatki) [inline railties/lib/rails/generators/rails/app/templates/config/databases/oracle.yml]: @matthewd @schneems Found this PR which removed `ENV["DATABASE_URL"]` from production configuration :)

Comment (prathamesh-sonpatki) [inline railties/lib/rails/generators/rails/app/templates/config/databases/oracle.yml]: @schneems I was referring to this comment in our discussion - https://github.com/rails/rails/pull/14633#issuecomment-39840551

Comment (matthewd) [inline railties/lib/rails/generators/rails/app/templates/config/databases/oracle.yml]: 🙈

Comment (schneems) [inline railties/lib/rails/generators/rails/app/templates/config/databases/oracle.yml]: Lol. Glad we caught up in person. We can re-add the `DATABASE_URL` generation in config/database.yml. I'm still worried about it being taken out (again).

**Pull request merged**